InteractionBase/src/vis/ViewSwitcherMenu.cpp

```diff
@@ -96,13 +118,19 @@ void ViewSwitcherMenu::endFocusMode(Visualization::Item *target)
 
 bool ViewSwitcherMenu::executeEntry(Visualization::Item* item)
 {
+	//static int nameCounter = 0;
 	if (inEditMode_) return false;
 	if (auto entry = DCast<VViewSwitcherEntry>(item))
 	{
 		QPoint pos = indexOf(item);
 		auto view = scene()->viewItems()->viewItem(entry->nameField()->text());
 		if (!view)
-			view = scene()->viewItems()->newViewItem(entry->nameField()->text(), pos);
+		{
+			auto name = entry->nameField()->text();
+			if (name == "Empty slot")
+				name = "View" + QString::number(pos.x()) + QString::number(pos.y()) + "";//QString::number(nameCounter++);
```

Comment:
So, this is much better than before, but we can still get in trouble (e.g. If I manually called some other view using the name that would be autogenerated for this one). We need to make sure that any automatically generated name is unique. For that matter, any manually entered name should also be rejected if it already exists. I'm actually fine with your original approach that just uses a static counter. Simple and does the job. We will still need to prevent manually creating duplicates views though.